src/main/scala/devsearch/blobToAST/ASTExtractor.scala

```diff
@@ -0,0 +1,151 @@
+package devsearch.blobToAST
+
+import devsearch.ast.{StringSource, ContentsSource, AST}
+import devsearch.parsers._
+import org.apache.spark.rdd.RDD
+import org.apache.spark.{SparkConf, SparkContext}
+import scala.util.parsing.combinator._
+
+
+/**
+ * Created by hubi on 3/27/15.
+ */
+abstract class CodeFile extends java.io.Serializable
+  case class JavaFile(size: String, owner: String, repository: String, path: String, code: String)
+    extends CodeFile
+  case class PythonFile(size: String, owner: String, repository: String, path: String, code: String)
+    extends CodeFile
+  case class GoFile(size: String, owner: String, repository: String, path: String, code: String)
+    extends CodeFile
+  case class JavaScriptFile(size: String, owner: String, repository: String, path: String, code: String)
+    extends CodeFile
+  case class UnknownFile() extends CodeFile
+
+
+object SnippetParser extends RegexParsers with java.io.Serializable {
+  def parseBlob: Parser[CodeFile] = (
+    number~":Java/"~noSlash~"/"~noSlash~"/"~path~code ^^ {
+      case size~_~owner~_~repo~_~path~code => JavaFile(size, owner, repo, path, code)
+    }
+    |number~":Python/"~noSlash~"/"~noSlash~"/"~path~code ^^ {
+      case size~_~owner~_~repo~_~path~code => PythonFile(size, owner, repo, path, code)
+    }
+    |number~":Go/"~noSlash~"/"~noSlash~"/"~path~code ^^ {
+      case size~_~owner~_~repo~_~path~code => GoFile(size, owner, repo, path, code)
+    }
+    |number~":JavaScript/"~noSlash~"/"~noSlash~"/"~path~code ^^ {
+      case size~_~owner~_~repo~_~path~code => JavaScriptFile(size, owner, repo, path, code)
+    }
+  )
+
+  val number:  Parser[String] = """\d+""".r
+  val noSlash: Parser[String] = """[^/]+""".r
+  val path:    Parser[String] = """[^\n]+""".r                 //everything until eol
+  val code:    Parser[String] = """(?s).*[^\n\d+:]""".r        //everything until "\n897162346:"
```

Comment:
Indeed, the parse rule is wrong, I didn't notice that.
Especially since the `toBlobSnippet` function uses the length while parsing, so this is actually useless. The code is simply the rest of the string. I'll push a correction in [devsearch-learning](https://github.com/devsearch-epfl/devsearch-learning).